[WEB-1364] Integrations page fixes

bgdeutsch opened the pull request:
### What does this PR do?
Integrations page front end fixes
- `All` category filter should be highlighted if there is no category filter in the URL hash, or if the hash is `#all`.
- Filtering works when there is a valid query parameter present in the URL (i.e. `/?q=azure`)

### Motivation
https://datadoghq.atlassian.net/browse/WEB-1364

### Preview
- https://docs-staging.datadoghq.com/brian.deutsch/integrations-query-param/integrations/?q=mesos
  - Filtering based on query parameter should work correctly.

- https://docs-staging.datadoghq.com/brian.deutsch/integrations-query-param/integrations/?q=azure#cat-messaging
  - When both query param and category hash are present, filtering should be based on query param.  Category filter falls back to `#all`.

- https://docs-staging.datadoghq.com/brian.deutsch/integrations-query-param/integrations/#cat-marketplace
  - Filtering with a category hash in the URL still works as expected.

- Integrations page should otherwise work as expected.

- Synthetics: https://dd-corpsite.datadoghq.com/synthetics/details/5zm-7b4-ifw

### Additional Notes
<!-- Anything else we should know when reviewing?-->

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

davidejones commented:
This looks great but I noticed that entering in a new text search doesn't update the url. 

e.g
- visit https://docs-staging.datadoghq.com/brian.deutsch/integrations-query-param/integrations/?q=mesos
- enter adobe in text input
- copy url in browser to share with a friend and this will be the wrong url

bgdeutsch commented:
> This looks great but I noticed that entering in a new text search doesn't update the url.
> 
> e.g
> 
> * visit https://docs-staging.datadoghq.com/brian.deutsch/integrations-query-param/integrations/?q=mesos
> * enter adobe in text input
> * copy url in browser to share with a friend and this will be the wrong url

hey @davidejones i've addressed this so user input in the search bar will be reflected in the URL query param.   i've also made another update so that if there is a query param present, and user clicks one of the category filter buttons that isn't `All`, the query param will be stripped out.   added some support for those two changes in my synthetics too: https://dd-corpsite.datadoghq.com/synthetics/details/5zm-7b4-ifw.   thanks!
